x-pack/plugins/infra/public/pages/metrics/hosts/components/page_providers.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { HostsTableProvider } from '../hooks/use_hosts_table';
+import { HostsViewProvider, useHostsViewContext } from '../hooks/use_hosts_view';
+import { useUnifiedSearchContext } from '../hooks/use_unified_search';
+
+export const HostsTableStateProvider = ({ children }: { children: React.ReactNode }) => {
+  const { hostNodes } = useHostsViewContext();
+  const { searchCriteria } = useUnifiedSearchContext();
+
+  return (
+    <HostsTableProvider nodes={hostNodes} tableParams={{ time: searchCriteria.dateRange }}>
+      {children}
+    </HostsTableProvider>
+  );
+};
+export const PageProviders = ({ children }: { children: React.ReactNode }) => {
+  return (
+    <HostsViewProvider>
+      <HostsTableStateProvider>{children}</HostsTableStateProvider>
+    </HostsViewProvider>
+  );
+};
```

Comment:
**question**: there is any reason behind creating this wrapper component instead of directly consuming the `useHostsViewContext` and `useUnifiedSearchContext` into the `useHostsTable` implementation?

Reply:
good question. I guess my first reason was to avoid nesting context providers in [`hosts_container.tsx`](https://github.com/elastic/kibana/pull/155249/files#diff-0b848aa74af22f252256dd1754f67524746b04f79a9cc06f1a89ff0637df3029R40). There are benefits in doing that if you memoize the provider values - which is not the case here. I will change this to consume `useUnifiedSearchContext` directly in `useHostTable`